modules/web/js/ballerina/diagram/views/default/components/connector-properties.jsx

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import ConnectorHelper from '../../../../env/helpers/connector-helper';
+import './properties-form.css';
+import ASTFactory from '../../../../ast/ast-factory';
+import PropertiesWindow from './property-window';
+/**
+ * React component for a connector prop window
+ *
+ * @class ConnectorPropertiesForm
+ * @extends {React.Component}
+ */
+class ConnectorPropertiesForm extends React.Component {
+
+    constructor(props) {
+        super(props);
+        this.getSupportedProps = this.getSupportedProps.bind(this);
+        this.getDataAddedToConnectorInit = this.getDataAddedToConnectorInit.bind(this);
+        this.setDataToConnectorInitArgs = this.setDataToConnectorInitArgs.bind(this);
+        this.getConnectorInstanceString = this.getConnectorInstanceString.bind(this);
+        this.getAddedValueOfProp = this.getAddedValueOfProp.bind(this);
+        this.getFullPkgPath = this.getFullPkgPath.bind(this);
+    }
+
+    /**
+     * Get already added values to properties
+     * @param node
+     * @returns {string}
+     */
+    getAddedValueOfProp(node) {
+        let value = '';
+        if (ASTFactory.isBasicLiteralExpression(node)) {
+            value = node.getBasicLiteralValue();
+        } else if (ASTFactory.isSimpleVariableReferenceExpression(node)) {
+            value = node.getVariableName();
+        }
+        return value;
+    }
+
+    /**
+     * Get full package path of connector
+     * @param pkgName
+     * @returns {string}
+     */
+    getFullPkgPath(pkgName) {
```

Comment:
We should get this dynamically.

Reply:
This was hard coded because we cannot get the full package path of the connector from the composer design view atm. This is an issue with the fragment parser and can be resolved with the implementation of the new semantic analyser.